### PR TITLE
feat: Listing the Assigned Tags on the MemberDetail screen

### LINKED ISSUE #2424: ADMIN: Show the list of tags assigned to a member on their details page.
### Is your feature request related to a problem? Please describe.

Listing the tags assigned to a user, the functionality to navigate to an assigned tag, and unassigning it.

### Additional Context

previous issue: #2395

### PR DESCRIPTION
**Pull request by meetulr**
### What kind of change does this PR introduce?

Lists the tags assigned to a user. Adds to functionality to navigate to that tag and unassign it.

### Issue Number:

Fixes #2424

### Did you add tests for your changes?

Yes

### Demo

https://github.com/user-attachments/assets/1945b1be-e599-4d7a-aa29-331dc0c3a567

## Release Notes

- **New Features**
  - Enhanced user tag management with new messages for unassigning tags and feedback on tag assignments.
  - Integrated modal for unassigning user tags within the Member Detail screen.
  - Improved pagination for fetching assigned tags in user detail queries.
  - Added localization support for user tag management in multiple languages.

- **Bug Fixes**
  - Addressed minor issues in translation files for better consistency and clarity.

- **Style**
  - Updated styles for improved layout and interactivity in the Member Detail component.

- **Tests**
  - Comprehensive enhancements to the test suite for the Member Detail component, including new tests for tag management and improved test organization.

- **Chores**
  - Refactored mock data for user details and tag management to support better testing scenarios.

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
       - name: Count number of lines
         run: |
           chmod +x ./.github/workflows/countline.py
-          ./.github/workflows/countline.py --lines 600 --exclude_files src/screens/LoginPage/LoginPage.tsx src/GraphQl/Queries/Queries.ts src/screens/OrgList/OrgList.tsx src/GraphQl/Mutations/mutations.ts src/components/EventListCard/EventListCardModals.tsx src/components/TagActions/TagActionsMocks.ts src/utils/interfaces.ts
+          ./.github/workflows/countline.py --lines 600 --exclude_files src/screens/LoginPage/LoginPage.tsx src/GraphQl/Queries/Queries.ts src/screens/OrgList/OrgList.tsx src/GraphQl/Mutations/mutations.ts src/components/EventListCard/EventListCardModals.tsx src/components/TagActions/TagActionsMocks.ts src/utils/interfaces.ts src/screens/MemberDetail/MemberDetail.tsx
```
[the remark this answers is not in the document]
Comment (meetulr) [reply .github/workflows/pull-request.yml:41]: will look at refactoring later.

File: src/screens/MemberDetail/MemberDetail.test.tsx [hunk 2]
```diff
@@ -311,30 +52,54 @@ jest.mock('@mui/x-date-pickers/DateTimePicker', () => {
...
-        <BrowserRouter>
-          <Provider store={store}>
-            <I18nextProvider i18n={i18nForTest}>
-              <MemberDetail {...props} />
-            </I18nextProvider>
-          </Provider>
-        </BrowserRouter>
-      </MockedProvider>,
-    );
+    await wait();
```
[the remark this answers is not in the document]
Comment (meetulr) [reply src/screens/MemberDetail/MemberDetail.test.tsx:100]: It's just waiting for the render to be completed.

File: src/screens/MemberDetail/MemberDetail.test.tsx [hunk 3]
```diff
@@ -1,308 +1,49 @@
...
+import { toast } from 'react-toastify';
 
 const link1 = new StaticMockLink(MOCKS1, true);
 const link2 = new StaticMockLink(MOCKS2, true);
 const link3 = new StaticMockLink(MOCKS3, true);
 
-async function wait(ms = 20): Promise<void> {
+async function wait(ms = 500): Promise<void> {
   await act(() => new Promise((resolve) => setTimeout(resolve, ms)));
 }
```
[the remark this answers is not in the document]
Comment (meetulr) [reply src/screens/MemberDetail/MemberDetail.test.tsx:31]: We're waiting for some time after the renders. Will further look into it later.

File: src/screens/MemberDetail/MemberDetail.tsx [hunk 4]
```diff
@@ -122,6 +140,73 @@ const MemberDetail: React.FC<MemberDetailProps> = ({ id }): JSX.Element => {
...
+                ],
+                pageInfo: fetchMoreResult.user.user.tagsAssignedWith.pageInfo,
+                totalCount: fetchMoreResult.user.user.tagsAssignedWith.pageInfo,
+              },
+            },
+          },
+        };
+      },
+    });
+  };
```
[the remark this answers is not in the document]
Comment (meetulr) [reply src/screens/MemberDetail/MemberDetail.tsx:178]: That's just how the query is structured right now, will have to make modifications later.

File: src/screens/MemberDetail/MemberDetail.tsx [hunk 5]
```diff
@@ -521,41 +607,134 @@
...
+                        </Button>
+                      </div>
+                      {index + 1 !== tagsAssigned.length && (
+                        <hr className="mx-0" />
+                      )}
+                    </div>
+                  ))}
+                </InfiniteScroll>
+              )}
+            </Card.Body>
```
[the remark this answers is not in the document]
Comment (meetulr) [reply src/screens/MemberDetail/MemberDetail.tsx:682]: It's correct, the query name and the de-structured variable name for userData needs to be updated.

**Review by beingnoble03 — APPROVED** [review]
looks good!

**Pull request merged**